src/test/java/uk/gov/ons/census/action/model/repository/CaseRepositoryIT.java

```diff
@@ -0,0 +1,171 @@
+package uk.gov.ons.census.action.model.repository;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.data.jpa.domain.Specification.where;
+
+import java.util.*;
+import javax.persistence.criteria.CriteriaBuilder;
+import org.jeasy.random.EasyRandom;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.ons.census.action.model.entity.Case;
+
+@ActiveProfiles("test")
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class CaseRepositoryIT {
+
+  private static final String TEST_ACTION_PLAN_ID;
+  private static final Map<String, List<String>> TEST_CLASSIFIERS;
+  private EasyRandom easyRandom = new EasyRandom();
+
+  @Autowired private CaseRepository caseRepository;
+
+  static {
+    TEST_ACTION_PLAN_ID = "ddbd26c2-15af-4055-aab9-591b3735b8d3";
+
+    TEST_CLASSIFIERS = new HashMap<>();
+    TEST_CLASSIFIERS.put(
+        "treatmentCode",
+        Arrays.asList(
+            "HH_LF2R3AE",
+            "HH_LF3R3BE",
+            "HH_LF3R3AE",
+            "HH_LF2R1E",
+            "HH_LFNR2E",
+            "HH_LFNR3AE",
+            "HH_LF2R3BE",
+            "HH_LF3R2E",
+            "HH_LF2R2E",
+            "HH_LF3R1E"));
+  }
+
+  private void setupTenUnReceiptedCases() {
+    List<Case> unReceiptedCases = new ArrayList<>();
+    for (int i = 0; i < 10; i++) {
+      Case unReceiptedCase = easyRandom.nextObject(Case.class);
+      unReceiptedCase.setActionPlanId(TEST_ACTION_PLAN_ID);
+      unReceiptedCase.setReceiptReceived(false);
+      unReceiptedCases.add(unReceiptedCase);
+      unReceiptedCase.setTreatmentCode("HH_LF3R1E");
+    }
+    caseRepository.saveAll(unReceiptedCases);
+  }
+
+  private void setupThreeOfTenReceiptedCases() {
+    List<Case> unReceiptedCases = new ArrayList<>();
+    for (int i = 0; i < 7; i++) {
```

Comment:
This function could be joined with the 1 above. Looking something like
```
void addCasesToDB( int caseCount, boolean caseReceipted) {
    for (int i = 0; i < caseCount; i++) {
      Case unReceiptedCase = easyRandom.nextObject(Case.class);
      unReceiptedCase.setActionPlanId(TEST_ACTION_PLAN_ID);
      unReceiptedCase.setReceiptReceived(caseReceipted);
      unReceiptedCase.setTreatmentCode("HH_LF3R1E");
      caseRepositorty.save(unReceiptedCase);
    }
  }

```
Then call it with addCasesToDB( 7, false ) , addCasesToDB(3, true)

Reply:
done